### ISSUE TITLE
[Bug] Several models hardcode supported attention backends. Causing confusion

### ISSUE THREAD

**Issue by watbulb** [hao-ai-lab/FastVideo#1254]
### Describe the bug

In several model implementations, WAN, hunyuan, LTX, etc. you hardcode the supported attention backend. Is there a specific reason why?

It seems you have support for: `SAGE_ATTN`, `FLASH_ATTN`, `TORCH_SDPA`, `VIDEO_SPARSE_ATTN`,
  `VMOBA_ATTN`, `SAGE_ATTN_THREE`, `SLA_ATTN`, `SAGE_SLA_ATTN`, but `wanvideo.py:135` only lists `FLASH_ATTN`, `TORCH_SPDA`?

This causes several confusing logging conditions where it doesn't matter if you set, say SAGE_ATTN, it will interleave two different attention backends because different attention layers in the model declare different `supported_attention_backends`.

```
INFO 04-24 19:27:08.515 [cuda.py:117] Trying FASTVIDEO_ATTENTION_BACKEND=SAGE_ATTN
INFO 04-24 19:27:08.515 [cuda.py:118] Selected backend: None
INFO 04-24 19:27:08.546 [cuda.py:235] Cannot use FlashAttention-2 backend because the flash_attn package is not found. Make sure that flash_attn was built and installed (on by default).
INFO 04-24 19:27:08.546 [cuda.py:242] Using Torch SDPA backend.
INFO 04-24 19:27:08.546 [cuda.py:117] Trying FASTVIDEO_ATTENTION_BACKEND=SAGE_ATTN
INFO 04-24 19:27:08.546 [cuda.py:118] Selected backend: None
INFO 04-24 19:27:08.546 [cuda.py:235] Cannot use FlashAttention-2 backend because the flash_attn package is not found. Make sure that flash_attn was built and installed (on by default).
INFO 04-24 19:27:08.546 [cuda.py:242] Using Torch SDPA backend.
INFO 04-24 19:27:08.546 [cuda.py:117] Trying FASTVIDEO_ATTENTION_BACKEND=SAGE_ATTN
INFO 04-24 19:27:08.546 [cuda.py:118] Selected backend: None
INFO 04-24 19:27:08.546 [cuda.py:235] Cannot use FlashAttention-2 backend because the flash_attn package is not found. Make sure that flash_attn was built and installed (on by default).
INFO 04-24 19:27:08.546 [cuda.py:242] Using Torch SDPA backend.
INFO 04-24 19:27:08.546 [cuda.py:117] Trying FASTVIDEO_ATTENTION_BACKEND=SAGE_ATTN
INFO 04-24 19:27:08.546 [cuda.py:118] Selected backend: None
INFO 04-24 19:27:08.546 [cuda.py:235] Cannot use FlashAttention-2 backend because the flash_attn package is not found. Make sure that flash_attn was built and installed (on by default).
INFO 04-24 19:27:08.546 [cuda.py:242] Using Torch SDPA backend.
INFO 04-24 19:27:08.546 [cuda.py:117] Trying FASTVIDEO_ATTENTION_BACKEND=SAGE_ATTN
INFO 04-24 19:27:08.546 [cuda.py:118] Selected backend: None
INFO 04-24 19:27:08.546 [cuda.py:235] Cannot use FlashAttention-2 backend because the flash_attn package is not found. Make sure that flash_attn was built and installed (on by default).
INFO 04-24 19:27:08.546 [cuda.py:242] Using Torch SDPA backend.
```

### Reproduction

Run any basic inference example. 

### Environment

Irrelevant.

**Comment by alexzms** [comment]
Currently the model layers only wire up FLASH_ATTN, TORCH_SDPA, and VIDEO_SPARSE_ATTN. Plumbing other backends through is on the roadmap and contributions are very welcome!

**Comment by watbulb** [comment]
> Currently the model layers only wire up FLASH_ATTN, TORCH_SDPA, and VIDEO_SPARSE_ATTN. Plumbing other backends through is on the roadmap and contributions are very welcome!
> 

Okay, thanks for letting me know! I think I've been able to address this on a local fork along with some windows native patches if you're welcome to that.

EDIT: of course we have to be careful to choose backends that support masks or don't depending on the model selection which introduces a support matrix, but I think there is a natural way to detect whether a model is compatible with a selected attention backend.

**Comment by alexzms** [comment]
Thanks @watbulb! A PR is definitely welcome. Just keep in mind the design here probably needs some care — per-layer assumptions (mask requirements, shape/dtype, kv-cache) vary across backends, so it's easy for a refactor to silently regress one model. Happy to take a look once you have something.

**Comment by UtkarshRjn** [comment]
Following up — I opened #1486 as the small first step: it makes the silent fallback **explicit** (a warning naming the requested backend and the layer's supported set), which addresses the "no idea my choice was ignored" half of this issue without changing any behavior.

For the larger fix, I looked at how vLLM's `selector.py` (which this file was adapted from) handles this. vLLM keeps the selector thin and pushes capability checks into the platform, where each backend **self-describes** its constraints (e.g. `get_supported_head_sizes()`) and a validation chain logs a clear reason on every fallback. I'd like to mirror that here, incrementally:

1. ✅ Warn instead of silently nulling an unsupported choice (#1486).
2. Add capability self-description to `AttentionBackend` (e.g. `get_supported_head_sizes`, `get_supported_dtypes`, `supports_attention_mask`, `supports_varlen`, `is_available`) — purely additive, defaults preserve current behavior. This also lets the platform's `get_attn_backend_cls` drop its scattered `try/except ImportError` blocks.
3. Centralize the per-layer allowlists into one declarative place as the single source of truth, rather than hand-listing tuples across ~16 model files.
4. Only *enable* a new backend×model combination once a correctness test (SSIM/numerical) confirms it — so we capture @alexzms's point that the hardcoded lists encode tested-good combos, not just theoretical compatibility.

The key distinction I want to respect: vLLM's checks are pure hardware capability, but FastVideo layers also encode *semantic* appropriateness (e.g. VSA on DiT self-attention vs. dense on cross-attention). So the design keeps both — backend-declared capability **and** layer-declared needs — and never auto-enables a combo just because it type-checks.

Does this direction sound right to you before I start on step 2? Happy to scope each step as its own small PR.

**Comment by UtkarshRjn** [comment]
One addition for **step 3**: the source-of-truth should also carry an optional model-declared **required** backend, not just each layer's **supported** set.

Some models are architecturally tied to one backend. FastWan (#1146) is the clearest case — its block has a `to_gate_compress` projection that only exists in the VSA variant, so the checkpoint won't even load unless VSA is selected. That's a *requirement*, the mirror of the *allowlist* we already encode.

So step 3 would hold both: **supported** (layer may use) and **required** (model must use — forced when unset, clear error on conflict/unavailability). Reserving room for `required` lets #1146 land as a small follow-up instead of a one-off field we refactor in later. Sound reasonable @alexzms?
